### ISSUE TITLE
Vary width of day based on events

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#2591]
Originally reported on Google Code with ID 2326

```
I was wondering if adding a setting for half-width weekends in the month view would
be generally useful. For my use, 90% of the time there's nothing on the weekend, but
I don't want to hide them with

http://fullcalendar.io/docs/display/weekends/

because every now and again there will be an event. However, would it be possible to
show Saturdays and Sundays compressed, to allow more space for the rest of the week?
Attached is a sample calendar from our system. If Saturday and Sunday were half the
width, there'd be more space for text in the weekday events.

I don't know how hard this is, I imagine it could be complex, but no harm suggesting
it hopefully...

Oliver
www.agilebase.co.uk
```

Reported by `oliver.kohll` on 2014-10-20 18:26:11

<hr>
- _Attachment: calendar-colourful.png<br>![calendar-colourful.png](https://storage.googleapis.com/google-code-attachments/fullcalendar/issue-2326/comment-0/calendar-colourful.png)_

Imported with **1** stars.

**Comment by stavro** [comment]
> will keep this on the issue tracker and see if there is more demand for it

Definitely!  I've experimented a little with this, but if you have any thoughts around implementation I could also help to implement.

Thanks

**Comment by EugenVau** [comment]
+1

**Comment by espen** [comment]
@EugenVau see https://fullcalendar.io/wiki/Requesting-Features/ regarding upvoting features

**Comment by mickaelramanitrera** [comment]
I was able to workaround this by setting a specific width to first and last columns through css like this : 

.fc-day-number {
  float: left!important;
}

.fc-row.fc-week.fc-widget-content > .fc-bg > table > tbody > tr > td:first-child,
.fc-row.fc-week.fc-widget-content > .fc-bg > table > tbody > tr > td:last-child,
.fc-content-skeleton > table > tbody > tr > td:first-child,
.fc-content-skeleton > table > tbody > tr > td:last-child,
.fc-content-skeleton > table > thead > tr > td:first-child,
.fc-content-skeleton > table > thead > tr > td:last-child {
  width: 35px;
}`
`

**Comment by okohll** [comment]
@mickaelramanitrera Thanks for the pointer to using CSS. I was able to use the following with Fullcalendar v5:

```
#calendar.fc .fc-daygrid .fc-day-sat,
#calendar.fc .fc-daygrid .fc-day-sun {
  width: 8.333%; /* 1/12th i.e. the two weekend days fit into the same space as one of the other five days */
}
```

`#calendar` is the ID of the element Fullcalendar's applied to in my system, it will probably work without.

Oliver (the original requester of this issue)

**Comment by FredTSousa** [comment]
I need this but in order to show more events with more data on days with lots of overlapping events.

I'm doing some workarounds in order to make it work but its really hacky and i don't like it.
basically running this to all resources.

`var setWidthForResourcePerEvent = (resourceId) => {
            let index_highest=1,
                eventsByResource=document.querySelectorAll('[data-resource-id="'+resourceId+'"] .fc-timegrid-event-harness'),
                i;

            for(i=0; i<eventsByResource.length; i++) {
                cur = eventsByResource[i];
                var index_current = parseInt(getComputedStyle(cur)["z-index"], 10);
                 if(index_current > index_highest) {
                    index_highest = index_current;
                }
            }
            var resourcesToSize=document.querySelectorAll('[data-resource-id="'+resourceId+'"]');
            for(i=0; i<resourcesToSize.length; i++) {
                cur = resourcesToSize[i];
                resourcesToSize[i].style.width=(((fullCalendar.calendar.getOption('dayMinWidth')-8)*index_highest)+8)+"px";
            }
            document.querySelector('.fc-timegrid-slots > table').style.width=document.querySelector('table.fc-col-header.fc-scrollgrid-sync-table').style.width;
        }`

**Comment by Mihai-github** [comment]
+1

**Comment by paulwer** [comment]
current solution, would be to add the following classes:

.fc-day-sun,
.fc-day-sat {
  width: 7%;
}
